crates/biome_js_analyze/src/analyzers/nursery/no_unused_dependencies.rs

```diff
@@ -0,0 +1,56 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::JsModuleSource;
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow the use of dependencies that aren't specified in the `package.json`
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// import "vite";
+    /// ```
+    ///
+    pub(crate) NoUnusedDependencies {
+        version: "next",
+        name: "noUnusedDependencies",
+        recommended: false,
+    }
+}
+
```

Comment:
One question I wonder: should the rule checks dev and optional dependencies?

Reply:
The dependencies received are `dependencies` and `devDependencies`, although I should passing `PackageJson` would help to understand the distinction

Reply:
I think it might make sense to have an option for dev dependencies. Most of the time they shouldn’t be allowed, but you should generally make exceptions for test and/or dev scripts that reside in the repository. Of course it wouldn’t be a big deal to disable the rule for those either.

For optional dependencies it might make sense to only allow them in dynamic imports?

Reply:
> The dependencies received are `dependencies` and `devDependencies`.

I wouldn’t consider this a good default behavior, btw. An important reason to apply a rule like this, for library authors at least, is to make sure you don’t accidentally use dependencies that your users don’t have access to. Downstream users wouldn’t have the `devDependencies` installed, so the rule should report that.

Reply:
Thank you. I will follow up with an option